src/core/thread/mle.cpp

```diff
@@ -4469,5 +4487,22 @@ void Mle::UpdateAllDomainBackboneRouters(BackboneRouter::Leader::DomainPrefixSta
 }
 #endif // OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
 
+// Since Thread 1.2, MTD end device (including rx-on-when-idle or rx-off-when-idle) should
+// register or de-register their multicast address larger than realm local to the parent via
+// MLE Child Update Request.
+bool Mle::ShouldRegisterMulticastToParent(void) const
+{
+    bool update = false;
+
+    VerifyOrExit(IsChild() && !IsFullThreadDevice(), OT_NOOP);
+#if (OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_1)
+    VerifyOrExit(!IsRxOnWhenIdle(), OT_NOOP);
+#endif
+    update = true;
```

Comment:
If we choose the strategy that 1.2 MED does not send MA registration to 1.1 parent, should we check for 1.1 parent here?

Reply:
updated. please take a review.